File: services/src/data/clusters.ts
import { Collection } from 'mongodb';
import { db } from '../config/db-client';
import { Cluster } from '../models/cluster';

export class Clusters {
  private clusters: Collection<Cluster>;

  constructor() {
    this.clusters = db.collection('clusters') as Collection<Cluster>;
  }

  async findOne(id: string) {
    return await this.clusters.findOne({ id });
  }

  async find(contractAddress: string) {
    return await this.clusters.find({ contractAddress }).toArray();
  }
}

File: services/cluster-collection/src/data/clusters/clusters.py
from config.mongoDb import get_db
from dataclasses import asdict
from models.cluster import Cluster


class ClustersClass():
    def __init__(self):
        db = get_db()
        self.collection = db['clusters']

    def addClusters(self, clusters: list[Cluster], session) -> bool:
        serializedClusters = [asdict(c) for c in clusters]

        try:
            result = self.collection.insert_many(
                serializedClusters, session=session)

            return result.acknowledged
        except:
            print('Could not add clusters to DB')
            raise Exception('Could not add clusters to DB')
